optimum/habana/transformers/generation/utils.py

```diff
@@ -1856,11 +1859,11 @@ def _greedy_search(
 
             if ignore_eos:
                 this_peer_finished = stopping_criteria(
-                    input_ids, scores, token_idx=cur_len, ignore_eos=ignore_eos, eos_token_id=eos_token_id
+                    input_ids, scores, token_idx=token_idx, ignore_eos=ignore_eos, eos_token_id=eos_token_id
```

Comment:
stop_tkn_idx = token_idx if token_idx is None else cur_len

stopping_criteria(input_ids, scores, token_idx=stop_tkn_idx , ignore_eos=ignore_eos, eos_token_id=eos_token_id)